server/src/test/java/org/elasticsearch/rest/action/info/RestClusterInfoActionTests.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.rest.action.info;
+
+import org.elasticsearch.action.admin.cluster.node.stats.NodeStats;
+import org.elasticsearch.action.admin.cluster.node.stats.NodesStatsResponse;
+import org.elasticsearch.client.internal.node.NodeClient;
+import org.elasticsearch.cluster.ClusterName;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.http.HttpStats;
+import org.elasticsearch.http.HttpStatsTests;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.rest.FakeRestRequest;
+import org.junit.Before;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static org.elasticsearch.rest.action.info.RestClusterInfoAction.AVAILABLE_METRICS;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.hasToString;
+import static org.mockito.Mockito.mock;
+
+public class RestClusterInfoActionTests extends ESTestCase {
+
+    private RestClusterInfoAction action;
+
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        action = new RestClusterInfoAction();
+    }
+
+    public void testUnrecognizedMetric() {
+        var metric = randomAlphaOfLength(32);
+        var request = new FakeRestRequest.Builder(xContentRegistry()).withPath("/_info/").withParams(Map.of("metric", metric)).build();
+
+        var e = expectThrows(IllegalArgumentException.class, () -> action.prepareRequest(request, mock(NodeClient.class)));
+        assertThat(e, hasToString(containsString("request [/_info/] contains unrecognized metric: [" + metric + "]")));
+    }
+
+    public void testShouldNotMixAllWithOtherMetrics() {
+        var wrongMetric = randomAlphaOfLength(32);
+        var request = new FakeRestRequest.Builder(xContentRegistry()).withPath("/_info/")
+            .withParams(Map.of("metric", "_all," + wrongMetric))
+            .build();
+
+        var e = expectThrows(IllegalArgumentException.class, () -> action.prepareRequest(request, mock(NodeClient.class)));
+        assertThat(e, hasToString(containsString("request [/_info/] contains _all and individual metrics [_all," + wrongMetric + "]")));
+    }
+
+    public void testAllMetricsAlone() throws IOException {
+        var metric = "_all";
+        var request = new FakeRestRequest.Builder(xContentRegistry()).withPath("/_info/").withParams(Map.of("metric", metric)).build();
+
+        action.prepareRequest(request, mock(NodeClient.class));
```

Comment:
Seems less than ideal to not assert anything in this test (and the next). But I just looked into what it would take to assert something about the result and I see why you didn't go down that path! Think it's worth adding a yaml rest test that covers `_all`?

Reply:
yeah, yesterday I invested ~1h trying to figure out how to test it, to find almost no success scenario tests are doing asserts. I planned to add the `_all` in the next PR.